modules/integration/tests-integration/lazy-loading-test/src/test/java/org/wso2/appserver/integration/lazy/loading/LazyLoadingBaseTest.java

```diff
@@ -0,0 +1,486 @@
+        tenantIdleTime =
+                Long.parseLong(asServer.getConfigurationNode(TENANT_IDLE_XPATH).getNodeValue()) * 60 * 1000;
+        webAppIdleTime =
+                Long.parseLong(asServer.getConfigurationNode(WEB_APP_IDLE_XPATH).getNodeValue()) * 60 * 1000;
+        artifactsLocation =
+                TestConfigurationProvider.getResourceLocation() + File.separator + "artifacts" + File.separator +
+                        "AS" + File.separator + "ghost" + File.separator;
+        String tenantInfoServiceArtifactLocation =
+                artifactsLocation + TENANT_INFO_SERVICE_FILE_NAME;
+        String carbonArtifactLocation = artifactsLocation + CARBON_XML;
+        File sourceFile = new File(carbonArtifactLocation);
+        File targetFile = new File(CARBON_REPOSITORY_LOCATION);
+        superTenantDomain =
+                new AutomationContext(PRODUCT_GROUP_NAME, INSTANCE_NAME, SUPER_TENANT_DOMAIN_KEY, ADMIN_USER_KEY).
+                        getSuperTenant().getDomain();
+        tenantDomain1 =
+                new AutomationContext(PRODUCT_GROUP_NAME, INSTANCE_NAME, TENANT_DOMAIN_1_KEY, ADMIN_USER_KEY).
+                        getContextTenant().getDomain();
+        tenantDomain2 =
+                new AutomationContext(PRODUCT_GROUP_NAME, INSTANCE_NAME, TENANT_DOMAIN_2_KEY, ADMIN_USER_KEY).
+                        getContextTenant().getDomain();
+        serverManager.applyConfigurationWithoutRestart(sourceFile, targetFile, true);
+        log.info("carbon.xml replaced with :" + carbonArtifactLocation);
+        webAppAdminClient.warFileUplaoder(tenantInfoServiceArtifactLocation);
+        serverManager.restartGracefully();
+        log.info("Server Restarted after applying carbon.xml and tenant information utility web application");
+    }
+
+
+    /**
+     * This method will do a REST  GET call to the tenant-info-service (deployed by init() method) to get the tenant
+     * status information
+     *
+     * @param tenantDomain -  Domain Name of the tenant.
+     * @return TenantStatus - a JSON object that contains whether the configuration context is loaded or not.
+     * @throws LazyLoadingTestException - Exception throws when send the Get request and retrieve the JSON data from response.
+     */
+    protected TenantStatus getTenantStatus(String tenantDomain) throws LazyLoadingTestException {
+        TenantStatus tenantStatus;
+        String requestUrl = webAppURL + "/" + IS_TENANT_LOADED_METHOD_URL + "/" + tenantDomain;
+        try {
+            HttpResponse response = HttpURLConnectionClient.sendGetRequest(requestUrl, null);
+            JSONObject tenantStatusJSON = new JSONObject(response.getData());
+            tenantStatus =
+                    new TenantStatus(tenantStatusJSON.getJSONObject("TenantStatus").getBoolean("tenantContextLoaded"));
+        } catch (IOException ioException) {
+            String customErrorMessage = "IOException when sending the Get request to:" + requestUrl;
```

Comment:
@prabathabey - IMO having a more specific name for variable name make sense. This variable name clearly indicates it is a custom error message